include/swift/AST/Types.h

```diff
@@ -5686,6 +5687,12 @@ class OpenedArchetypeType final : public ArchetypeType,
   /// Retrieve the opened existential type
   Type getOpenedExistentialType() const;
```

Comment:
This function is a bit of a footgun, because even on a nested opened archetype it still returns the top-level existential type, rather than the erased type for this particular archetype. We probably want to do a quick audit of clients to ensure that they are only using this on the root archetype, and consider renaming the API (`getOpenedExistentialTypeAtRoot` or similar) or removing it entirely (clients can ask for the opened existential type on the generic environment, which is obviously shared amongst all archetypes in that opened existential).

Reply:
Noticed this too; I will look into it in a follow-up.